design/workitems.go

```diff
@@ -257,3 +257,21 @@ var _ = a.Resource("planner_backlog", func() {
 		a.Response(d.InternalServerError, JSONAPIErrors)
 	})
 })
+
+var _ = a.Resource("named_work_items", func() {
+	a.Parent("namedspaces")
+	a.BasePath("/workitems")
+	a.Action("show", func() {
+		a.Routing(
+			a.GET("/:wiNumber"),
+		)
+		a.Description("Retrieve a work item from the given number.")
+		a.Params(func() {
+			a.Param("wiNumber", d.Integer, "Number of the work item to show")
+		})
+		a.Response(d.MovedPermanently)
+		a.Response(d.BadRequest, JSONAPIErrors)
```

Comment:
I think we don't have this `BadRequest` response for this endpoint. In the `LookupIDByNamedSpaceAndNumber` function you only return a NotFound and an InternalError but don't call anything else that can potentially raise a BadRequest error.